docs/data/tree-view/simple-tree-view/customization/CustomContentTreeView.tsx

```diff
@@ -1,95 +1,74 @@
 import * as React from 'react';
-import clsx from 'clsx';
 import { styled } from '@mui/material/styles';
-import Typography from '@mui/material/Typography';
 import Box from '@mui/material/Box';
 import Avatar from '@mui/material/Avatar';
 import { SimpleTreeView } from '@mui/x-tree-view/SimpleTreeView';
+import { useTreeItem } from '@mui/x-tree-view/internals/useTreeItem';
 import {
-  TreeItem,
-  useTreeItemState,
-  TreeItemProps,
-  TreeItemContentProps,
-} from '@mui/x-tree-view/TreeItem';
+  TreeItemNextContent,
+  TreeItemNextIconContainer,
+  TreeItemNextTransitionGroup,
+  TreeItemNextLabel,
+  TreeItemNextProps,
+  TreeItemNextRoot,
+} from '@mui/x-tree-view/internals/TreeItemNext';
+import { TreeItemIcon } from '@mui/x-tree-view/internals/TreeItemIcon';
+import { TreeItemProvider } from '@mui/x-tree-view/internals/TreeItemProvider';
 
-const CustomContentRoot = styled('div')(({ theme }) => ({
-  '&': { padding: theme.spacing(0.5, 1) },
+const CustomTreeItemContent = styled(TreeItemNextContent)(({ theme }) => ({
+  padding: theme.spacing(0.5, 1),
 }));
 
-const CustomContent = React.forwardRef(function CustomContent(
-  props: TreeItemContentProps,
-  ref,
+const CustomTreeItem = React.forwardRef(function CustomTreeItem(
+  props: TreeItemNextProps,
+  ref: React.Ref<HTMLLIElement>,
 ) {
   const {
-    className,
-    classes,
-    label,
+    id,
     nodeId,
-    icon: iconProp,
-    expansionIcon,
-    displayIcon,
+    label,
+    children,
+    slots = {},
+    slotProps = {},
```

Comment:
For me `slots` is ourtool to comunicate with dev because we lack of context about what they are doing. Devs have the product context so they should pass directly their custom icons to the component in a more semantic friendly way.

```diff
- <TreeItemIcon status={status} slots={slots} />
+ <TreeItemIcon status={status} slots={{ collapsedIcon: CustomIcon }} />
```

I they need to customize them at the `CustomTreeItem` level, I would encourage them to be semantically closer to their usecase. For example the VSCode would be

```jsx
<CustomTreeItem nodeId="1" label="DatePicker">
  <CustomTreeItem nodeId="2" label="index.ts" fileType="ts" />
  <CustomTreeItem nodeId="3" label="DatePicker.tsx"  fileType="tsx">
  <CustomTreeItem nodeId="6" label="DatePicker.test.ts" fileType="test" />
</CustomTreeItem>
```

With the `CustomTreeItem` slots `endIcon` computed according to the `fileType` (or the label extension)

![image](https://github.com/mui/mui-x/assets/45398769/84b6b91f-5cda-45f3-a854-f4058455fc56)

Reply:
perfect :+1;